### ISSUE TITLE
Temporary script file is not recreated when language changes

### ISSUE THREAD

**Issue by juliangilbey** [spyder-ide/spyder#21565]
### Issue Report Checklist

* [x] Searched the [issues page](https://github.com/spyder-ide/spyder/issues?q=is%3Aissue) for similar reports
* [x] Read the relevant sections of the [Spyder Troubleshooting Guide](https://github.com/spyder-ide/spyder/wiki/Troubleshooting-Guide-and-FAQ) and followed its advice
* [x] Reproduced the issue after updating with ``conda update spyder`` (or ``pip``, if not using Anaconda)
* [ ] Could not reproduce inside ``jupyter qtconsole`` (if console-related)
* [x] Tried basic troubleshooting (if a bug/error)
    * [x] Restarted Spyder
    * [x] Reset preferences with ``spyder --reset``
    * [ ] Reinstalled the latest version of [Anaconda](https://www.anaconda.com/download/)
    * [x] Tried the other applicable steps from the Troubleshooting Guide
* [x] Completed the **Problem Description**, **Steps to Reproduce** and **Version** sections below

## Problem Description

A minor issue.

I started a fresh install of Spyder in one language, then changed to another language through the preferences.  The temporary file `temp.py` (created in `spyder/plugins/editor/plugin.py`, `__load_temp_file`) was not recreated in the new language, so remained in the original language until the file was manually deleted (and then automatically recreated).

### What steps reproduce the problem?

1. Open a fresh install of Spyder in some locale (or run `spyder --reset`).
2. In Preferences > Application > Advanced settings, change the Language to another supported language.
3. Restart Spyder with no open file windows, so the default `temp.py` file is opened.

### What is the expected output? What do you see instead?

I started in German (`LC_ALL=de_DE.UTF-8 LANG=de` - a bit heavy-handed but it did the job).  I then switched to English, restarted Spyder and the file was still in German.  It should have switched to English.

One could argue that a user might have modified this file intentionally, so it should not be automatically recreated.  It does say, though, that it is a temporary script file.

## Versions

* Spyder version: 5.5.0  (pip)
* Python version: 3.11.6 64-bit
* Qt version: 5.15.10
* PyQt5 version: 5.15.10
* Operating System: Linux-6.5.0-4-amd64-x86_64-with-glibc2.37

### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                     :  1.4.1 (OK)
chardet >=2.0.0                          :  5.2.0 (OK)
cloudpickle >=0.5.0                      :  2.2.1 (OK)
cookiecutter >=1.6.0                     :  1.7.3 (OK)
diff_match_patch >=20181111              :  20230430 (OK)
intervaltree >=3.0.2                     :  3.0.2 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1         :  8.14.0 (OK)
jedi >=0.17.2,<0.20.0                    :  0.18.2 (OK)
jellyfish >=0.7                          :  0.8.9 (OK)
jsonschema >=3.2.0                       :  4.10.3 (OK)
keyring >=17.0.0                         :  24.3.0 (OK)
nbconvert >=4.0                          :  6.5.3 (OK)
numpydoc >=0.6.0                         :  1.5.0 (OK)
parso >=0.7.0,<0.9.0                     :  0.8.3 (OK)
pexpect >=4.4.0                          :  4.8.0 (OK)
pickleshare >=0.4                        :  0.7.5 (OK)
psutil >=5.3                             :  5.9.5 (OK)
pygments >=2.0                           :  2.15.1 (OK)
pylint >=2.5.0,<3.1                      :  2.17.4 (OK)
pylint_venv >=3.0.2                      :  3.0.2 (OK)
pyls_spyder >=0.4.0                      :  0.4.0 (OK)
pylsp >=1.9.0,<1.10.0                    :  1.9.0 (OK)
pylsp_black >=1.2.0,<3.0.0               :  1.3.0 (OK)
qdarkstyle >=3.2.0,<3.3.0                :  3.2.1 (OK)
qstylizer >=0.2.2                        :  0.2.2 (OK)
qtawesome >=1.2.1                        :  1.2.3 (OK)
qtconsole >=5.5.0,<5.6.0                 :  5.5.1 (OK)
qtpy >=2.1.0                             :  2.4.1 (OK)
rtree >=0.9.7                            :  1.1.0 (OK)
setuptools >=49.6.0                      :  68.1.2 (OK)
sphinx >=0.6.6                           :  7.2.6 (OK)
spyder_kernels >=2.5.0,<2.6.0            :  2.5.0 (OK)
textdistance >=4.2.0                     :  4.5.0 (OK)
three_merge >=0.1.1                      :  0.1.1 (OK)
watchdog >=0.10.3                        :  3.0.0 (OK)
xdg >=0.26                               :  0.28 (OK)
zmq >=22.1.0                             :  24.0.1 (OK)

# Optional:
cython >=0.21                            :  0.29.36 (OK)
matplotlib >=3.0.0                       :  3.6.3 (OK)
numpy >=1.7                              :  1.24.2 (OK)
pandas >=1.1.1                           :  1.5.3 (OK)
scipy >=0.17.0                           :  1.10.1 (OK)
sympy >=0.7.3                            :  1.12 (OK)

# Spyder plugins:
spyder_line_profiler.spyder.plugin 0.3.2 :  0.3.2 (OK)
spyder_unittest.unittestplugin 0.6.0     :  0.6.0 (OK)
```

**Comment by dalthviz** [comment]
Hi @juliangilbey thank you for the feedback! So when changing the language and seeing the `temp.py` you were expecting to see it in the currently selected language, right? I guess that could make sense if the file is empty/has not been manually changed by the user 🤔

What do you think @spyder-ide/core-developers ?

**Comment by jitseniesen** [comment]
The request to re-create the temporary file in the new language does make sense to me, but there is a risk of deleting stuff that the user wanted to keep. I think I agree that it's best to re-create the temporary file only if it has not been changed.

**Comment by mrclary** [comment]
> One could argue that a user might have modified this file intentionally, so it should not be automatically recreated. It does say, though, that it is a temporary script file.

One thing that I do not like about the Editor is that it opens a `temp.py` file instead of an `untitled<N>.py` at startup or when switching/closing projects and no files are open. However, when closing all the editor tabs manually, an `untitled<N>.py` file opens. This is inconsistent and I think that `untitled<N>.py` should open in all scenarios when there is not an open file.

So, what is the purpose of the `.spyder-py3/temp.py` file? And why shouldn't Spyder open an `untitled<N>.py` file (copied from `.spyder-py3/template.py`) in all scenarios?

As far as languages are concerned, whether we are dealing with `.spyder-py3/temp.py` or `.spyder-py3/template.py`, why not create other language versions on demand? For example, if `.spyder-py3/temp-<lang>.py` does not already exist, create it (copied or translated from `.spyder-py3/temp.py`). `.spyder-py3/temp-<lang>.py` can be modified by the user if desired. If the user switches languages, then a new `temp-<lang>.py` file is created (if it does not already exist), but the previous language instance is not removed, i.e. `.spyder-py3/temp-en.py` and `.spyder-py3/temp-de.py` would both exist and never over-written by Spyder.

**Comment by dalthviz** [comment]
> One thing that I do not like about the Editor is that it opens a temp.py file instead of an untitled<N>.py at startup or when switching/closing projects and no files are open. However, when closing all the editor tabs manually, an untitled<N>.py file opens. This is inconsistent and I think that untitled<N>.py should open in all scenarios when there is not an open file.
>
>So, what is the purpose of the .spyder-py3/temp.py file? And why shouldn't Spyder open an untitled<N>.py file (copied from .spyder-py3/template.py) in all scenarios?

That's quite true (not sure why that works in that way to be honest :thinking:)

> As far as languages are concerned, whether we are dealing with .spyder-py3/temp.py or .spyder-py3/template.py, why not create other language versions on demand? For example, if .spyder-py3/temp-<lang>.py does not already exist, create it (copied or translated from .spyder-py3/temp.py).

I though also something similar as a first approach to prevent deleting something users defined from the `temp.py` file while having the initial default content in sync with the language preference :+1:

**Comment by ccordoba12** [comment]
> What do you think @spyder-ide/core-developers ?

To me this is a very minor issue that doesn't require too much bikeshedding: if the user changes the interface language, the temp file shouldn't be recreated (I mean, it's assumed they know how to speak the initial language in which Spyder started).

> So, what is the purpose of the .spyder-py3/temp.py file? And why shouldn't Spyder open an untitled.py file (copied from .spyder-py3/template.py) in all scenarios?
>> That's quite true (not sure why that works in that way to be honest, maybe to not start creating a lot of untitle files around or be prompt always to set the new file name and choose where to save it 🤔).

The idea is to allow users to start programming right away without the fear of losing their work if they close Spyder or it crashes unexpectedly.

Then I think to change that behavior we'd need to save all untitled files to disk and show them in the next session (unless users close them, of course). It shouldn't be that hard, so you're welcome to do that.

**Comment by juliangilbey** [comment]
I don't have any strong views about this one; whatever you feel makes sense is fine from my perspective!
